uber/models.py

```diff
@@ -825,7 +823,22 @@ def sorted_attendees(self):
         return self.attendees
 
     @property
+    def unassigned(self):
+        """
+        Returns a list of the unassigned badges for this group, sorted so that
+        the paid-by-group badges come last, because when claiming unassigned
+        badges we want to claim the "weird" ones first.
+        """
+        return sorted([a for a in self.attendees if a.is_unassigned], key=lambda a: a.paid == c.PAID_BY_GROUP)
+
+    @property
     def floating(self):
+        """
+        Returns the list of paid-by-group unassigned badges for this group. This
+        is a separate property from the "Group.unassigned" property because when
+        automatically adding or removing unassigned badges, we care specifically
+        about paid-by-group badges rather than all unassigned badges.
+        """
```

Comment:
I believe the docstrings explain the need and use for both properties, but let me know if this is unclear.